### PR TITLE
Change Settings for the Decimals to the Wins and Draw Results and Inc…

### PR DESCRIPTION
**Pull request by Dronatelis**

### CODE REVIEW

**Comment by Cernelius** [comment]
I'm not a developer so just wanted to make sure about a thing not knowing if it is already the case.

The approximation should not be done by truncation.

For example, with two decimals, if the result is 79.996...%, it should be given as 80.00%, not 79.99%.

Other than that, I don't have an opinion on whether (for example) something like 0.005% at two decimals should be approximated to 0.01% or to 0.00%. If TripleA is based in the United States of America, I would go with whatever the prevailing convention there is (if anyone knows).

This would be relevant to decide, for example, if I'm running 200 calculations and win only once, whether I should be told I've 1% or 0% chances to win (0.5% approximated to integer).

Reference:
https://en.wikipedia.org/wiki/Rounding#Rounding_to_the_nearest_integer

I realize that this almost irrelevant in practice.

(Sorry if this is already clear.)

**Comment by Cernelius** [comment]
Personally, beside rounding, I would not round whenever the number is exactly half-way between the two closest possible approximations. For example, at two decimals,
- 79.9951% becomes 80.00%.
- 79.9949% becomes 79.99%.
- 79.995% remains 79.995%.

This implies that, at X decimals, one will actually have X+1 decimals whenever having a situation which would otherwise require some tie-breaking rule in order to be approximated at X decimals.

(I've no idea if this is feasible.)

Again, I realize that this is almost irrelevant in practice.

---

Of course, decimals can be added too. For example, at two decimals,
- 80.1% becomes 80.10%.
- 80% becomes 80.00%.
- 79.9% becomes 79.90%.

**Comment by asvitkine** [comment]
> I think i have done it but i will have to run some tests. I am little busy i will make request tomorow if thats OK

Are you still planning to make updates here?

**Comment by Cernelius** [comment]
I'm not a developer, but I guess that, if this already improves the current situation, I'd rather merge it than close it if no more work is getting done even if it is only half-baked and does not address each of the points to improve. Up to you to judge.

File: game-app/game-core/src/main/java/games/strategy/triplea/odds/calculator/BattleCalculatorPanel.java [hunk 1]
```diff
@@ -676,8 +680,32 @@ private Territory findPotentialBattleSite() {
...
+  private static String formatPercentage(final double percentage, int numberOfRounds) {
+    String pattern;
+    if (numberOfRounds < 10_000) {
+      pattern = "0%.##";
+    } else if (numberOfRounds < 100_000) {
+      pattern = "0%.0#";
+    } else {
+      pattern = "0%.00";
+    }
+
```
Comment (asvitkine) [inline game-app/game-core/src/main/java/games/strategy/triplea/odds/calculator/BattleCalculatorPanel.java:702]: Remove empty line.

File: game-app/game-core/src/main/java/games/strategy/triplea/odds/calculator/BattleCalculatorPanel.java [hunk 2]
```diff
@@ -676,8 +680,32 @@ private Territory findPotentialBattleSite() {
...
+//    } else if (numberOfRounds < 100_000) {
+//      return new DecimalFormat("0%.0#").format(percentage);
+//    } else {
+//      return new DecimalFormat("0%.00").format(percentage);
+//    }
+//  }
+
+  private static String formatPercentage(final double percentage, int numberOfRounds) {
+    String pattern;
+    if (numberOfRounds < 10_000) {
```
Comment (asvitkine) [inline game-app/game-core/src/main/java/games/strategy/triplea/odds/calculator/BattleCalculatorPanel.java:695]: Add a comment explaining why the pattern is based on the number of rounds.

File: game-app/game-core/src/main/java/games/strategy/triplea/odds/calculator/BattleCalculatorPanel.java [hunk 3]
```diff
@@ -676,8 +680,32 @@ private Territory findPotentialBattleSite() {
     return location;
   }
 
-  private static String formatPercentage(final double percentage) {
-    return new DecimalFormat("#%.##").format(percentage);
+//  private static String formatPercentage(final double percentage, int numberOfRounds) {
```
Comment (asvitkine) [inline game-app/game-core/src/main/java/games/strategy/triplea/odds/calculator/BattleCalculatorPanel.java:683]: Remove commented out code...

File: game-app/game-core/src/main/java/games/strategy/triplea/odds/calculator/BattleCalculatorPanel.java [hunk 4]
```diff
@@ -676,8 +680,32 @@ private Territory findPotentialBattleSite() {
...
+      pattern = "0%.0#";
+    } else {
+      pattern = "0%.00";
+    }
+
+
+    DecimalFormat decimalFormat = new DecimalFormat(pattern);
+    decimalFormat.setRoundingMode(RoundingMode.HALF_UP); // Round the value
+    System.out.println(percentage);
+    System.out.println(decimalFormat.format(percentage));
```
Comment (asvitkine) [inline game-app/game-core/src/main/java/games/strategy/triplea/odds/calculator/BattleCalculatorPanel.java:707]: Remove print statements.

File: game-app/game-core/src/main/java/games/strategy/triplea/odds/calculator/BattleCalculatorPanel.java [hunk 5]
```diff
@@ -676,8 +680,32 @@ private Territory findPotentialBattleSite() {
...
+    return decimalFormat.format(percentage);
   }
```
Comment (asvitkine) [inline game-app/game-core/src/main/java/games/strategy/triplea/odds/calculator/BattleCalculatorPanel.java:709]: Can you add some unit tests for this function?

**Comment by Cernelius** [comment]
Sup @Dronatelis ?

**Comment by asvitkine** [comment]
@Dronatelis Just wondering if you're planning to come back to this to address the review comments? Let me know if there's anything unclear...

**Comment by asvitkine** [comment]
@Dronatelis Do you still plan to work on this, or do you prefer someone else picks this up?

**Pull request closed without merging**